### PR TITLE
[DCOS-44001] Fix GitHub Security Warnings

### PR DESCRIPTION
**Pull request by kaiwalyajoshi**
Update the following packages:
- cryptography to 2.3.1
- requests to 2.20.0

Remove pycrypto-2.6.1 by removing python3-crypto
Replace pycrypto-2.6.1 with pycryptodome-3.7.0

### CODE REVIEW

**Comment by kaiwalyajoshi** [comment]
Seems mesosphere/dcos-commons-base will need to be updated, checking with team on update procedures.

**Comment by kaiwalyajoshi** [comment]
Breaking this up into multiple fixes.

https://github.com/mesosphere/dcos-commons/pull/2784 for dcos-commons-base fix.

File: Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # not be updating this dependency.  See
 # https://mesosphere.slack.com/archives/C4E91G0CX/p1541505296001800 for more
 # background.
-FROM mesosphere/dcos-commons-base@sha256:e9f67d72aa1431f4a15544981285f028decc7507538f8e5aa8b0888a94d820c0
+FROM mesosphere/dcos-commons-base:af8469627a8cc4c823fac826a5e02f0acaf07869
```
Comment (elezar) [inline Dockerfile]: Is this not a step in the wrong direction with regards to managing these dependencies?

Comment (elezar) [reply Dockerfile]: Does this meant that you built and pushed this image separately?

File: frozen_requirements.txt [hunk 2]
```diff
@@ -85,7 +85,7 @@ pyasn1==0.4.4
 pyasn1-modules==0.2.2
 pycodestyle==2.3.1
 pycparser==2.19
-pycrypto==2.6.1
+pycryptodome==3.7.0
```
Comment (elezar) [inline frozen_requirements.txt]: Would this not require changes to `test_requirements.txt` too?

File: Dockerfile (hunk 1 above)
Comment (kaiwalyajoshi) [reply Dockerfile]: @elezar that was the strategy used here, this PR builds the dcos-commons-base image: https://github.com/mesosphere/dcos-commons/pull/2784

Here we are pointing to the built image, with the other updates.

I've pushed up a commit to use the sha256 hash as opposed to the git-hash which was used before for pulling the correct version of the image.

File: frozen_requirements.txt (hunk 2 above)
Comment (kaiwalyajoshi) [reply frozen_requirements.txt]: I'm not fully clear on how the security vulnerabilities apply to test_requirements.txt. test_requirements.txt doesn't have any mentions of installing via the base-image, specifically python3-crypto. I could use help here if I'm missing something.

Comment (elezar) [reply frozen_requirements.txt]: I mean that we currently have two different "requirements" files that could now be out of sync due to the switch from `pycrypto` to `pycryptodome`. (but looking at the file there isn't anything specified there)

**Pull request merged**